xiilib/_gunicorn/charm.py

```diff
@@ -59,12 +61,21 @@ def __init__(self, framework: ops.Framework, wsgi_framework: str) -> None:
             self._update_app_and_unit_status(ops.BlockedStatus(exc.msg))
             return
 
+        requires = load_requires()
+        if "redis" in requires and requires["redis"]["interface"] == "redis":
```

Comment:
Should this be wrapped into a `try ... except` in case of a malformed `charmcraft.yaml`? Not sure if that would be caught by charmcraft before the build step

Reply:
specifically, for example, if `redis` is defined but perhaps doesn't have the `interface` key?